include/merlin_hashtable.cuh

```diff
@@ -579,6 +579,230 @@ class HashTable {
+                            table_value_ptr + n, param_key_index_ptr,
+                            thrust::less<uintptr_t>());
+      }
+
+      if (options_.io_by_cpu) {
+        const size_type host_ws_size{
+            dev_ws_size + n * (sizeof(bool) + sizeof(value_type) * dim())};
+        auto host_ws{host_mem_pool_->get_workspace<1>(host_ws_size, stream)};
+        auto h_table_value_addrs{host_ws.get<value_type**>(0)};
+        auto h_param_key_index{reinterpret_cast<int*>(h_table_value_addrs + n)};
+        auto h_founds{reinterpret_cast<bool*>(h_param_key_index + n)};
+        auto h_param_values{reinterpret_cast<value_type*>(h_founds + n)};
+
+        CUDA_CHECK(cudaMemcpyAsync(h_table_value_addrs, d_table_value_addrs,
+                                   dev_ws_size, cudaMemcpyDeviceToHost,
+                                   stream));
+        CUDA_CHECK(cudaMemcpyAsync(h_founds, founds, n * sizeof(bool),
+                                   cudaMemcpyDeviceToHost, stream));
+        CUDA_CHECK(cudaMemcpyAsync(h_param_values, values,
+                                   n * sizeof(value_type) * dim(),
+                                   cudaMemcpyDeviceToHost, stream));
+        CUDA_CHECK(cudaStreamSynchronize(stream));
+
+        read_or_write_by_cpu<value_type>(h_table_value_addrs, h_param_values,
+                                         h_param_key_index, h_founds, dim(), n);
+        CUDA_CHECK(cudaMemcpyAsync(values, h_param_values,
+                                   n * sizeof(value_type) * dim(),
+                                   cudaMemcpyHostToDevice, stream));
+      } else {
+        const size_t block_size = options_.io_block_size;
+        const size_t N = n * dim();
+        const size_t grid_size = SAFE_GET_GRID_SIZE(N, block_size);
+
+        read_or_write_kernel<key_type, value_type, meta_type>
+            <<<grid_size, block_size, 0, stream>>>(
+                d_table_value_addrs, values, founds, param_key_index, dim(), N);
+      }
+    }
+
+    CudaCheckError();
+  }
+
+  /**
+   * @brief Assign new key-value-meta tuples into the hash table.
+   * If the key doesn't exist, the operation on the key will be ignored.
+   *
+   * @param n Number of key-value-meta tuples to insert or assign.
+   * @param keys The keys to insert on GPU-accessible memory with shape
+   * (n).
+   * @param values The values to insert on GPU-accessible memory with
+   * shape (n, DIM).
+   * @param metas The metas to insert on GPU-accessible memory with shape
+   * (n).
+   * @parblock
+   * The metas should be a `uint64_t` value. You can specify a value that
+   * such as the timestamp of the key insertion, number of the key
+   * occurrences, or another value to perform a custom eviction strategy.
+   *
+   * The @p metas should be `nullptr`, when the LRU eviction strategy is
+   * applied.
+   * @endparblock
+   *
+   * @param stream The CUDA stream that is used to execute the operation.
+   */
+  void assign(const size_type n,
```

Comment:
Hi @WonderingWJ , please help confirm the API define is matching your requirement, thank you~